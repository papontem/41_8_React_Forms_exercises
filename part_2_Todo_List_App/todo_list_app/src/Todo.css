.Todo-container {
	border: 1px solid gold;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
}
.Todo {
	
	outline: 1px solid red;
}
