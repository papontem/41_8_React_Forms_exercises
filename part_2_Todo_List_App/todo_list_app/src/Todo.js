/*
    - Displays a div with the task of the todo.
    - Includes a button with the text "X" that, when clicked, removes the todo.
    - Optionally, may include additional functionality for editing or marking as complete
*/
function Todo(props) {
    const { task, deleteTodo} = props;
    console.log("Task:",task);

    const handleDeleteClick = () => {
		deleteTodo();
	};

    return (
		<div className="Todo-container">
			<div className="Todo">
				{/* I AM A TOOOODOOOOOOOO */}
                <p>{task}</p>
			</div>
			<button onClick={handleDeleteClick}>X</button>
		</div>
	);
}

export default Todo;