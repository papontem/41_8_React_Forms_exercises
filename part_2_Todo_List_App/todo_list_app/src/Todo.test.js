import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For expect assertions
import Todo from "./Todo";

test("renders box without crashing", () => {
	render(<Todo />);
});

test("Todo snapshot", () => {
	// Render the component
	const { asFragment } = render(<Todo />);

	// Take a snapshot of the rendered component
	expect(asFragment()).toMatchSnapshot();
});
