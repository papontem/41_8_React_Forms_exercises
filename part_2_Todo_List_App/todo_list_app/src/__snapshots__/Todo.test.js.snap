// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo-container"
  >
    <div
      class="Todo"
    >
      <p />
    </div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
